### PR TITLE
Update jekyll.yml

### PR DESCRIPTION
**Pull request by MOJOliciousFTW**
## GitHub Copilot Summary
This pull request includes updates to the Jekyll site deployment workflow to ensure it runs on both push and pull request events, and adds conditional steps for pull requests. The most important changes include adding the `pull_request` trigger, and adding conditions to the build and deploy steps.

Improvements to workflow triggers and conditions:

* [`.github/workflows/jekyll.yml`](diffhunk://#diff-60ac246cc02819dea1007076ba57241caa7c7d1e8232a5179124d4e63bb366f8L10-R13): Added `pull_request` trigger to run the workflow on pull requests targeting the master branch.
* [`.github/workflows/jekyll.yml`](diffhunk://#diff-60ac246cc02819dea1007076ba57241caa7c7d1e8232a5179124d4e63bb366f8L30-R68): Added condition to the `Upload artifact` step to skip it during pull request events.
* [`.github/workflows/jekyll.yml`](diffhunk://#diff-60ac246cc02819dea1007076ba57241caa7c7d1e8232a5179124d4e63bb366f8L30-R68): Added condition to the `deploy` job to skip it during pull request events and specified that it needs the `build` job to complete.

## Breaking change

Don't know, maybe.

## Proposed change

Build on PR, but no upload of artifact nor deploy job.

Intent is to build on PR to catch build errors before merge to master. Did I get this right? Or will this change mess something up?

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.

### CODE REVIEW

**Comment by dschaele** [comment]
The way it built and excluded the deploy https://github.com/frdedynamics/hvl_robotics_website/actions/runs/10925925037 looks promising. I guess this should work

**Comment by dschaele** [comment]
any other concenrs @MOJOliciousFTW ?

File: .github/workflows/jekyll.yml [hunk 1]
```diff
@@ -27,37 +28,44 @@ concurrency:
...
       - name: Setup Ruby
         uses: ruby/setup-ruby@f321cf5a4d1533575411f8752cf25b86478b0442 # v1.193.0
         with:
           ruby-version: '3.1' # Not needed with a .ruby-version file
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
           cache-version: 0 # Increment this number if you need to re-download cached gems
+
       - name: Setup Pages
         id: pages
         uses: actions/configure-pages@v5
```
Comment (MOJOliciousFTW) [inline .github/workflows/jekyll.yml:47]: I guess `actions/configure-pages` doesn't hurt from `actions/deploy-pages` being cancelled. The "However, do NOT cancel in-progress runs as we want to allow these production deployments to complete." on line 25 threw me off 😅

File: .github/workflows/jekyll.yml [hunk 2]
```diff
@@ -27,37 +28,44 @@ concurrency:
...
+
       - name: Build with Jekyll
         # Outputs to the './_site' directory by default
         run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
```
Comment (MOJOliciousFTW) [inline .github/workflows/jekyll.yml:51]: did a minimal effort check, this just builds. The `--baseurl` got me curios.

**Comment by dschaele** [comment]
Nice 💯 now we just need more pull requests hehe

**Pull request merged**